lib/iris/cube.py

```diff
@@ -772,6 +772,23 @@ def __init__(self, data, standard_name=None, long_name=None,
             for cell_measure, dims in cell_measures_and_dims:
                 self.add_cell_measure(cell_measure, dims)
 
+        # When True indexing may result in a view onto the original data array,
+        # to avoid unnecessary copying.
+        self._share_data = False
+
+    @property
+    def share_data(self):
+        """Share cube data when slicing/indexing cube if True."""
+        return self._share_data
+
+    @share_data.setter
+    def share_data(self, value):
+        # Realise the data if is hasn't already been as sharing lazy data is
+        # not right now possible or a usecase understood.
+        if self.has_lazy_data():
+            _ = self.data
```

Comment:
hello @bjlittle 

I explicitly asked for this to be included, as part of my review of the code, so I am more culpable than @cpelley 

I think that you are right that this should only be implemented for setting to True
I think that a better docstring would help

I have created #2584, pointed at 1.13.x, as a proposed 'bug fix' for these concerns